src/gwt/src/org/rstudio/studio/client/workbench/views/files/Files.java

```diff
@@ -1137,6 +1147,9 @@ private void renameFile(FileSystemItem file)
          return;
       
       inputPending_ = true;
+            
+      // notify listeners that this file is about to be renamed
+      eventBus_.fireEvent(new RenameFileInitiatedEvent(file.getPath()));
```

Comment:
This is a bit awkward. What we really want here is for the actual presentation of the rename dialog to be sequenced after any potential save has happened, but this isn't easily handled here. In practice, this will probably still give us enough time to save before the user has finished interacting with the dialog.